server/routerlicious/packages/services/src/throttlerHelper.ts

```diff
@@ -92,23 +92,25 @@ export class ThrottlerHelper implements IThrottlerHelper {
         };
     }
 
-    private getAmountToReplenishOnCooldown(throttlingMetric: IThrottlingMetrics, now: number): number {
+    private getTokenReplenishAmount(throttlingMetric: IThrottlingMetrics, now: number): number {
         const timeSinceLastCooldownInMs = now - throttlingMetric.lastCoolDownAt;
         // replenish "tokens" at most once per minCooldownInterval
         if (timeSinceLastCooldownInMs > this.minCooldownIntervalInMs) {
-            return Math.floor(timeSinceLastCooldownInMs / this.rate);
+            return Math.floor(timeSinceLastCooldownInMs * this.rateInOperationsPerMs);
```

Comment:
I think this should be timeSinceLastCooldownInMs / this.rateInOperationsPerMs

Reply:
The goal here is to calculate how many tokens to be replenished. `timeSinceLastCooldownInMs / this.rateInOperationsPerMs` would return with the units `ms^2 / token`. Instead, we want `tokens`
```
tsc = <time since cooldown>
ops = <operations/tokens>

               <ops>        
<tsc> ms   x   ——————   =   <tsc> x <ops> 
                1 ms
```

Reply:
This did get me thinking that there is no upper limit here, so the cooldown could theoretically fill too many tokens if enough time passes between requests. I'll add a param for a token limit